uhabits-android/src/main/java/org/isoron/uhabits/activities/common/dialogs/WeekdayPickerDialog.kt

```diff
@@ -73,7 +74,10 @@ class WeekdayPickerDialog :
             .setNegativeButton(
                 android.R.string.cancel
             ) { _: DialogInterface?, _: Int -> dismiss() }
-        return builder.create()
+
+        val dialog = builder.create()
+        dialog.dismissCurrent()
+        return dialog
```

Comment:
After we return this dialog, does the other class call `dialog.show()`? If so, I think we should move `dismissCurrent` there.

Reply:
In this case, DialogFragment.show() is called, so I don't really have access to the dialog until onCreateDialog() of this AppCompatDialogFragment is finished. That's why I put this code here.

Reply:
I figured it out in the end, but it required making separate methods for handling a regular Dialog and DialogFragment